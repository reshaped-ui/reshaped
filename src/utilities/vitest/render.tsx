import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
