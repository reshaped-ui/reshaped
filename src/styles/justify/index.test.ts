import { expect, test, describe } from "vitest";
import getJustifyStyles from "./index";

describe("Styles/Justify", () => {
	test("handles value", () => {
		expect(getJustifyStyles("start")).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getJustifyStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getJustifyStyles({ s: "start", m: "center", l: "end" })).toMatchSnapshot();
	});
});
