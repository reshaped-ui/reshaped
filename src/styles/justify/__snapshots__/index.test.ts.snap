// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Justify > handles responsive value 1`] = `
{
  "variables": {
    "--rs-justify-l": "end",
    "--rs-justify-m": "center",
    "--rs-justify-s": "start",
  },
}
`;

exports[`Styles/Justify > handles undefined value 1`] = `null`;

exports[`Styles/Justify > handles value 1`] = `
{
  "variables": {
    "--rs-justify-s": "start",
  },
}
`;
